tests/python/test_tensor_reflection.py

```diff
@@ -37,7 +42,72 @@ def test_unordered():
     m = 7
     p = 11
 
-    ti.root.dense(ti.k, n).dense(ti.i, m).dense(ti.j, p).place(val)
+    blk1 = ti.root.dense(ti.k, n)
+    blk2 = blk1.dense(ti.i, m)
+    blk3 = blk2.dense(ti.j, p)
+    blk3.place(val)
+
+    assert val.data_type() == ti.i32
+    assert val.shape() == (n, m, p)
+    assert val.dim() == 3
+    assert val.snode().parent(0) == val.snode()
+    assert val.snode().parent() == blk3
+    assert val.snode().parent(1) == blk3
+    assert val.snode().parent(2) == blk2
+    assert val.snode().parent(3) == blk1
+    assert val.snode().parent(4) == ti.root
+
+    assert val.snode() in blk3.get_children()
+    assert blk3 in blk2.get_children()
+    assert blk2 in blk1.get_children()
+    assert blk1 in ti.root.get_children()
+
+    expected_repr = f'ti.root => dense {[n]} => dense {[n, m]}' \
+        f' => dense {[n, m, p]} => place {[n, m, p]}'
+    assert repr(val.snode()) == expected_repr
+
+
+@ti.all_archs
+def test_unordered_matrix():
+    val = ti.Matrix(3, 2, ti.i32)
+
+    n = 3
+    m = 7
+    p = 11
+
+    blk1 = ti.root.dense(ti.k, n)
+    blk2 = blk1.dense(ti.i, m)
+    blk3 = blk2.dense(ti.j, p)
+    blk3.place(val)
 
     assert val.dim() == 3
     assert val.shape() == (n, m, p)
+    assert val.data_type() == ti.i32
+    assert val.loop_range().snode().parent(0) == val.loop_range().snode()
+    assert val.loop_range().snode().parent() == blk3
+    assert val.loop_range().snode().parent(1) == blk3
+    assert val.loop_range().snode().parent(2) == blk2
+    assert val.loop_range().snode().parent(3) == blk1
+    assert val.loop_range().snode().parent(4) == ti.root
+
+
+@ti.all_archs
+def _test_var_parent():  # doesn't work :(
+    val1 = ti.var(ti.i32)
+    val2 = ti.var(ti.i32)
+    val3 = ti.var(ti.i32)
+
+    n = 3
+    m = 7
+    p = 11
+
+    blk1 = ti.root.dense(ti.k, n)
+    blk1.place(val1)
+    blk2 = blk1.dense(ti.i, m).place(val2)
+    blk3 = blk2.dense(ti.j, p)
+    blk3.place(val3)
+
+    assert val3.parent() == val3
```

Comment:
```suggestion
    assert val3.parent() == blk3
```

This part looks confusing.